docs/src/snap/howto/cis-hardening.md

```diff
@@ -282,6 +347,3230 @@ the sets, including the dqlite specific checks in the output.
+argument from this file so
+that the default takes effect.
+
+
+**Audit (as root):**
+
+```
+/bin/ps -ef | grep kube-apiserver | grep -v grep
+```
+
+**Expected output:**
+
+```
+--service-account-lookup is not set
+```
+
+##### Control 1.2.24
+
+**Description:**
+
+Ensure that the --service-account-key-file argument is set
+as appropriate (Automated)
+
+
+**Remediation:**
+
+Edit the API server configuration file /var/snap/k8s/common/args/kube-apiserver
+on the control plane node and set the --service-account-key-
+file parameter
+to the public key file for service accounts.
+
+`--service-account-key-file=<filename>`
+
+
+**Audit (as root):**
+
+```
+/bin/ps -ef | grep kube-apiserver | grep -v grep
+```
+
+**Expected output:**
+
+```
+--service-account-key-
+file=/etc/kubernetes/pki/serviceaccount.key
+```
+
+##### Control 1.2.25
+
+**Description:**
+
+Ensure that the --etcd-certfile and --etcd-keyfile arguments
+are set as appropriate (Automated)
+
+
+**Remediation:**
+
+Not applicable. Canonical K8s uses dqlite and the
+communication to this service is done through a
+local socket
+(/var/snap/k8s/common/var/lib/k8s-dqlite/k8s-dqlite.sock)
+accessible to users with root permissions.
+
+
+##### Control 1.2.26
+
+**Description:**
+
+Ensure that the --tls-cert-file and --tls-private-key-file
+arguments are set as appropriate (Automated)
+
+
+**Remediation:**
+
+Follow the Kubernetes documentation and set up the TLS
+connection on the apiserver. Then, edit the API server configuration
+file /var/snap/k8s/common/args/kube-apiserver
+on the control plane node and set the TLS certificate and
+private key file parameters.
+
+```
+--tls-cert-file=<path/to/tls-certificate-file>
+--tls-private-key-file=<path/to/tls-key-file>
+```
+
+
+**Audit (as root):**
+
+```
+/bin/ps -ef | grep kube-apiserver | grep -v grep
+```
+
+**Expected output:**
+
+```
+--tls-cert-file=/etc/kubernetes/pki/apiserver.crt and --tls-
+private-key-file=/etc/kubernetes/pki/apiserver.key
+```
+
+##### Control 1.2.27
+
+**Description:**
+
+Ensure that the --client-ca-file argument is set as
+appropriate (Automated)
+
+
+**Remediation:**
+
+Follow the Kubernetes documentation and set up the TLS
+connection on the apiserver. Then, edit the API server configuration
+file /var/snap/k8s/common/args/kube-apiserver
+on the control plane node and set the client certificate
+authority file.
+
+`--client-ca-file=<path/to/client-ca-file>`
+
+
+**Audit (as root):**
+
+```
+/bin/ps -ef | grep kube-apiserver | grep -v grep
+```
+
+**Expected output:**
+
+```
+--client-ca-file=/etc/kubernetes/pki/client-ca.crt
+```
+
+##### Control 1.2.28
+
+**Description:**
+
+Ensure that the --etcd-cafile argument is set as appropriate
+(Automated)
+
+
+**Remediation:**
+
+Not applicable. Canonical K8s uses dqlite and the
+communication to this service is done through a
+local socket
+(/var/snap/k8s/common/var/lib/k8s-dqlite/k8s-dqlite.sock)
+accessible to users with root permissions.
+
+
+**Expected output:**
```

Comment:
Do we need an expected output if we say it is not applicable?

Reply:
We do not. Good catch.

Reply:
See https://github.com/canonical/k8s-snap/pull/802/files#diff-da8b48161c214d6854a37a2ef8a664491413a6b43793ba6e9e6319a49da719ecR1748-R1763